### PR TITLE
backport to 1.16: udp: properly handle truncated/dropped datagrams

### PR DESCRIPTION
**Pull request by cpakulski**
Commit Message:
properly handle truncated/dropped datagrams

Also fixes examples to use v3.

Additional Description:
Risk Level: Med
Testing:Unit tests
Docs Changes: No
Release Notes: Yes

### CODE REVIEW

File: docs/root/version_history/current.rst [hunk 1]
```diff
@@ -1,5 +1,5 @@
-1.16.1 (Pending)
-================
+1.16.1 (November 20, 2020)
```
Comment (mattklein123) [inline docs/root/version_history/current.rst:1]: Note all of these need to get merged to main at some point, so please make a note on that.

File: examples/cors/backend/front-envoy.yaml [hunk 2]
```diff
@@ -8,13 +8,13 @@ static_resources:
     - filters:
       - name: envoy.filters.network.http_connection_manager
         typed_config:
-          "@type": type.googleapis.com/envoy.config.filter.network.http_connection_manager.v2.HttpConnectionManager
+          "@type": type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager
```
Comment (mattklein123) [inline examples/cors/backend/front-envoy.yaml:11]: It would be better if this change is a separate PR but not a crisis.

**Review by mattklein123 — APPROVED** [review]
Thanks!

File: examples/cors/backend/front-envoy.yaml (hunk 2 above)
Comment (cpakulski) [reply examples/cors/backend/front-envoy.yaml:11]: OK. Created #14125, which to be merged to release/v1.16 first and I will create a new PR for UDP fix.

Comment (cpakulski) [reply examples/cors/backend/front-envoy.yaml:11]: Created yet another PR #14127 for @antoniovicente fix to auth coverage test. I am going to close this PR, wait until the other two are merged to release/v1.16 and create a new PR just for UDP fixes.

**Comment by cpakulski** [comment]
Closing: this PR is replaced by #14130.

**Pull request closed without merging**